File: csharp/DVSClient/DVSClient/address/layout/CreateLayoutResult.cs
using DVSClient.Common;
using DVSClient.Server.Address.Layout;

namespace DVSClient.Address.Layout
{
    public class CreateLayoutResult
    {
        public ResponseError? Error { get; }
        public string Id { get; }

        public CreateLayoutResult(RestApiCreateLayoutResponse response)
        {
            Error = response.Error != null ? new ResponseError(response.Error) : null;
            Id = response.Result?.Id ?? string.Empty;
        }
    }
}

File: typescript/src/address/layout/createLayoutResult.ts
import { RestApiCreateLayoutResponse } from "../../server/address/layout/restApiCreateLayoutResponse";

export type CreateLayoutResult = {
    id: string;
    referenceId? : string,
};

export function restApiResponseToCreateLayoutResult(response: RestApiCreateLayoutResponse): CreateLayoutResult {
    return {id: response.result?.id??"", referenceId: response.referenceId,}
}